neuralop/training/callbacks.py

```diff
@@ -375,82 +376,123 @@ def on_before_loss(self, out):
     
     def on_before_val_loss(self, **kwargs):
         return self.on_before_loss(**kwargs)
-
-class ModelCheckpointCallback(Callback):
-
-    def __init__(self, checkpoint_dir: Union[Path, str] = Path('./checkpoints'), interval: int = 1):
-        """
-        Implements basic model checkpointing by saving a model every N epochs.
         
+class StateCheckpointCallback(Callback):
+    
+    def __init__(self, 
+                 save_dir: Union[Path, str] = './checkpoints', 
```

Comment:
I'm not sure it's a good idea to have this default, we could just make it a mandatory arg.